sky/serve/replica_managers.py

```diff
@@ -165,6 +177,34 @@ def _get_resources_ports(task_yaml: str) -> str:
     return task_resources.ports[0]
 
 
+def _get_use_spot_override(task_yaml: str,
+                           resource_override: Optional[Dict[str, Any]]) -> bool:
+    """Get whether the task uses spot from either overeride or task_yaml."""
+    if resource_override is not None:
+        use_spot_override = resource_override.get('use_spot')
+        if use_spot_override is not None:
+            assert isinstance(use_spot_override, bool)
+            return use_spot_override
+    task = sky.Task.from_yaml(task_yaml)
+    spot_use_resources = [
+        resources for resources in task.resources if resources.use_spot
+    ]
+    # Either resources all use spot or none use spot.
+    assert len(spot_use_resources) in [0, len(spot_use_resources)]
```

Comment:
Tried this YAML and it does not thrown any error too:

```yaml
service:
  readiness_probe: /health
  replicas: 1

resources:
  # use_spot: true
  ports: 8080
  any_of:
    - use_spot: true
    - use_spot: true
    - use_spot: false
    - use_spot: true
    - use_spot: false
```

Reply:
We should check this locally as well. Otherwise it will always `assert False` when it tries to provision a replica

Reply:
added. I think this should only apply when spot_policy is specified.